File: packages/app/src/services/software/definitions/itch.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'itch';
	public name = 'itch';
	public category = [SoftwareCategory.Gaming];
	public downloadName = 'itch-setup.exe';
	public icon = 'itch.png';
	public homepage = 'https://itch.io/app';

	public async resolveDownloadUrl() {
		return ok('https://itch.io/app/download');
	}
}

File: src/lib/software/itch.py
from src.lib.software import BaseSoftware

class Itch(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'itch-setup.exe'

    def resolve_download_url(self):
        self.url_resolved.emit('https://itch.io/app/download')
